### PR TITLE
rego-v1: Future-proofing `compiler` pkg tests

### PR DESCRIPTION
**Pull request by johanfylling**
Future-proofing `compiler` package tests to continue working when we flip the default rego-version to `v1` in OPA 1.0.

Also updating a few places that depended on `v0`-by-default assumptions.

### CODE REVIEW

File: compile/compile_test.go
Comment (ashutosh-narkar) [inline compile/compile_test.go:476]: Nit: remove this

File: compile/compile_test.go
Comment (ashutosh-narkar) [inline compile/compile_test.go:1127]: The generated policy is v1-only?

File: compile/compile_test.go
Comment (ashutosh-narkar) [inline compile/compile_test.go:2009]: Why not use `rego.v1` here and then the default rego version?

**Review by ashutosh-narkar — COMMENTED** [review]
I'm would have expected more test changes but this is good 😄  One question: We're explicitly using `regoV0` or `regoV1` in some test cases. I would have expected to import `rego.v1` in all cases and then just use the default Rego version which is `v0` for now in the test cases. Maybe I'm missing something.

File: compile/compile_test.go
Comment (johanfylling) [reply compile/compile_test.go:1127]: No, but the compiler will have resolved/stripped any imports in the optimized module, so we need to compare it with a parsed module that has no imports. Could just as well have been compared against a pure v0 module, but since we're moving towards v1 anyways, I opted for that.

Of course, I could also have set `WithRegoVersion(ast.RegoV1)` on the compiler, and not use the import. Maybe I should do that instead 🤔, and not mix apples and oranges ..

**Comment by johanfylling** [comment]
I've tried to use the `rego.v1` import as much as possible. For some cases, it's not possible because of the exact thing being tested, but for some, it would have been a huge change, and I took the lazy way out 😅.

Don't worry about the small amount of updated tests here 😉, I'm currently moving my way through the `ast` package compiler; that's _way_ bigger. Wouldn't surprise me if that's where we're going to find most of the affected tests. 
I'm putting up PRs in chunks, to not overload the reviewer.

File: compile/compile_test.go
Comment (johanfylling) [reply compile/compile_test.go:2010]: I hadn't added `ast.DefaultRegoVersion` yet when I was updating this test. Have fixed it now.

File: compile/compile_test.go
Comment (ashutosh-narkar) [reply compile/compile_test.go:1127]: That would make sense.

**Comment by ashutosh-narkar** [comment]
> Don't worry about the small amount of updated tests here 😉, I'm currently moving my way through the ast package compiler; that's way bigger.

That's what I was thinking about when I wrote that comment 🙂

**Comment by johanfylling** [comment]
I've got another PR lined up for the `ast` compiler. It builds on top of this PR, though, so I'll hold off on pushing it until this is merged.

File: compile/compile_test.go
Comment (ashutosh-narkar) [inline compile/compile_test.go:2657]: I imagine there's a reason for explicitly specifying the rego version and not just use the default. There are more instances of the same in the other places. So that's probably on purpose but please check before merging.

**Review by ashutosh-narkar — APPROVED** [review]

**Pull request merged**
